api/installations_service.go

```diff
@@ -0,0 +1,138 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"strings"
+)
+
+const (
+	StatusRunning   = "running"
+	StatusSucceeded = "succeeded"
+	StatusFailed    = "failed"
+)
+
+type InstallationsService struct {
+	client httpClient
+}
+
+type InstallationsServiceOutput struct {
+	ID     int
+	Status string
+	Logs   string
+}
+
+func NewInstallationsService(client httpClient) InstallationsService {
+	return InstallationsService{
+		client: client,
+	}
+}
+
```

Comment:
The command is in charge of orchestrating the services in what we've currently established. If I pushed the logic of retires down into each service, I would then have to have that logic coded separately for each of the three functions I have. Bringing the orchestration up to the highest level makes it more clear how the interplay of retires happens across all of the the services and that means the services do get to be mostly dumb about what is happening above them.